narwhals/_pandas_like/expr.py

```diff
@@ -387,19 +387,66 @@ def alias(self, name: str) -> Self:
         )
 
     def over(self, keys: list[str]) -> Self:
-        def func(df: PandasLikeDataFrame) -> list[PandasLikeSeries]:
-            if self._output_names is None:
-                msg = (
-                    "Anonymous expressions are not supported in over.\n"
-                    "Instead of `nw.all()`, try using a named expression, such as "
-                    "`nw.col('a', 'b')`\n"
+        cumulative_functions_to_pandas_equivalent = {
+            "col->cum_sum": "cumsum",
+            "col->cum_min": "cummin",
+            "col->cum_max": "cummax",
+            "col->cum_prod": "cumprod",
+            # Pandas cumcount starts counting from 0 while Polars starts from 1
+            # Pandas cumcount counts nulls while Polars does not
+            # So, instead of using "cumcount" we use "cumsum" on notna() to get the same result
+            "col->cum_count": "cumsum",
+        }
+
+        if self._function_name in cumulative_functions_to_pandas_equivalent:
+
+            def func(df: PandasLikeDataFrame) -> list[PandasLikeSeries]:
+                if self._output_names is None:
+                    msg = (
+                        "Anonymous expressions are not supported in over.\n"
+                        "Instead of `nw.all()`, try using a named expression, such as "
+                        "`nw.col('a', 'b')`\n"
+                    )
+                    raise ValueError(msg)
+                if self._root_names is None:
+                    msg = (
+                        "Anonymous expressions are not supported in over with cumulative "
+                        "operations.\n"
+                    )
+                    raise ValueError(msg)
+
+                if self._function_name == "col->cum_count":
+                    df = df.with_columns(~df[self._root_names[0]].is_null())
+
+                res_native = (
+                    df._native_frame.groupby(list(keys), as_index=False)[
+                        self._root_names[0]
+                    ]
+                    .transform(
+                        cumulative_functions_to_pandas_equivalent[self._function_name]
+                    )
+                    .rename(
+                        columns={self._root_names[0]: self._output_names[0]}, copy=False
+                    )
```

Comment:
Currently as a single column is passed, the result is a series and `rename(columns=...)` raises:

> TypeError: Series.rename() got an unexpected keyword argument ...

I think it is still doable to keep it for generic multiple expressions:

```suggestion
                res_native = (
                    df._native_frame.groupby(list(keys), as_index=False)[
                        self._root_names
                    ]
                    .transform(
                        cumulative_functions_to_pandas_equivalent[self._function_name]
                    )
                    .rename(
                        columns=dict(zip(self._root_names, self._output_names), copy=False
                    )
```

Reply:
also, `rename` should be imported from `_pandas_like/utils.py`